src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -115,6 +119,27 @@ public void OpenPin(int pinNumber, PinMode mode)
             SetPinMode(pinNumber, mode);
         }
 
+        /// <summary>
+        /// Opens a pin and sets it to a specific mode and value.
+        /// </summary>
+        /// <param name="pinNumber">The pin number in the controller's numbering scheme.</param>
+        /// <param name="mode">The mode to be set.</param>
+        /// <param name="initialValue">The initial value to be set if the mode is output. The driver will attempt to set the mode without causing glitches to the other value.
+        /// (if <paramref name="initialValue"/> is <see cref="PinValue.High"/>, the pin should not glitch to low during open)</param>
+        public void OpenPin(int pinNumber, PinMode mode, PinValue initialValue)
+        {
+            OpenPin(pinNumber);
+            if (GetPinMode(pinNumber) == PinMode.Output)
```

Comment:
Seems like we need an else clause here so that in pinmode was input, we still save the initial value so that when the mode is changed to output, this initial value is respected. Seems like the way it is now, we are only caring about initial value when mode is output.

Reply:
The whole condition can be removed, we just need to set the desired value.